shared-libs/search/src/search.js

```diff
@@ -40,7 +40,11 @@ module.exports = function(Promise, DB) {
     intersection = _.uniq(intersection, 'id');
     _.each(responses, function(response) {
       intersection = _.reject(intersection, function(row) {
-        return !_.findWhere(response, { id: row.id });
+        const existent = _.findWhere(response, { id: row.id });
```

Comment:
Non-actionable thought: It's a real bummer the response is an array instead of a map keyed by id...  We do so many seeks.

Reply:
Oh yea. Actionable, though. A more performant way would be to:
a. sort all arrays by docId
b. iterate over all lists once and at once, generating the intersection.

_warning, ugly code incoming_:

```
const getIntersection = (responses) => {
  // sort all responses by id
  responses = responses.map(response => response.sort((a,b) => (a.id > b.id) ? 1 : ((b.id > a.id) ? -1 : 0)));

  const intersection = [],
        ids = {};

  const indexes = Array(responses.length).fill(0);

  // helpers
  const getLowestIdIndex = values => {
    let min = values[0].id,
        minIdx = 0;
    values.forEach((value, index) => {
      if (value && value.id < min) {
        min = value.id;
        minIdx = index;
      }
    });

    return minIdx;
  };

  while (indexes.every((idx, arrIndex) => idx < responses[arrIndex].length)) {
    const values = indexes.map((idx, arrIndex) => responses[arrIndex][idx]);
    // if all responses have this ID
    if (_.uniq(values, 'id').length === 1) {
      const row = values.pop();
      if (!ids[row.id]) {
        // only push to the intersection if it doesn't have this ID yet
        values.forEach(value => {
          if ('sort' in value) {
            row.sort = value.sort + ' ' + (row.sort || row.value);
          }
        });
        intersection.push(row);
        ids[row.id] = 1;
      }

      // increases all indexes
      indexes.forEach((idx, i) => indexes[i]++);
      continue;
    }

    // only increase the lowest index
    indexes[getLowestIdIndex(values)]++;
  }

  return intersection;
};
```

For the few tests I did, it came out much faster.

Reply:
How much faster? :)

Reply:
I am _maybe_ considering it, do you think it's a blocker for this PR though? 
I'm gonna do a jsperf for it anyway :)

Reply:
No. I assumed it was not actionable feedback.

Reply:
If not blocking, I can create a separate issue to track this improvement.
Also, these are the perf results: 
https://jsperf.com/medic-search-intersection-reports & https://jsperf.com/medic-search-intersection